src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java

```diff
@@ -0,0 +1,155 @@
+                break;
+
+            case GET_TOPIC:
+                doGetTopic(routingContext);
+                break;
+
+            default:
+                throw new IllegalArgumentException("Unknown Operation: " + this.httpBridgeContext.getOpenApiOperation());
+
+        }
+    }
+
+    public void doListTopics(RoutingContext routingContext) {
+        listTopics(listTopicsResult -> {
+            if (listTopicsResult.succeeded()) {
+                JsonArray root = new JsonArray();
+                Set<String> topics = listTopicsResult.result();
+                topics.forEach(topic -> {
+                    root.add(topic);
+                });
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.OK.code(), BridgeContentType.KAFKA_JSON, root.toBuffer());
+            } else {
+                HttpBridgeError error = new HttpBridgeError(
+                        HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        listTopicsResult.cause().getMessage()
+                );
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            }
+        });
+    }
+
+    public void doGetTopic(RoutingContext routingContext) {
+        String topicName = routingContext.pathParam("topicname");
+
+        Future<Map<String, TopicDescription>> describeTopicsFuture = Future.future();
+        this.describeTopics(Collections.singletonList(topicName), describeTopicsFuture);
+        Future<Map<ConfigResource, Config>> describeConfigsFuture = Future.future();
+        this.describeConfigs(Collections.singletonList(new ConfigResource(org.apache.kafka.common.config.ConfigResource.Type.TOPIC,
+                topicName)), describeConfigsFuture);
+
+        CompositeFuture.join(describeTopicsFuture, describeConfigsFuture).onComplete(done -> {
+            if (done.succeeded() && describeTopicsFuture.result() != null && describeConfigsFuture.result() != null) {
+                Map<String, TopicDescription> topicDescriptions = describeTopicsFuture.result();
+                Map<ConfigResource, Config> configDescriptions = describeConfigsFuture.result();
+                JsonObject root = new JsonObject();
+                JsonArray partitionsArray = new JsonArray();
+                root.put("name", topicName);
+                List<ConfigEntry> configEntries = configDescriptions.values().iterator().next().getEntries();
+                if (configEntries.size() > 0) {
+                    JsonObject configs = new JsonObject();
+                    configEntries.forEach(configEntry -> {
+                        configs.put(configEntry.getName(), configEntry.getValue());
+                    });
+                    root.put("configs", configs);
+                }
+                TopicDescription description = topicDescriptions.values().iterator().next();
```

Comment:
as an additional check we can do something like this:

```java
TopicDescription description = topicDescriptions.get(topicName);
```

Reply:
ok. I switched to use get(topicName) and added an NPE guard.